proxy/src/registry.rs

```diff
@@ -595,6 +622,69 @@ impl Client for Registry {
         ))
     }
 
+    async fn transfer_from_user(
+        &self,
+        author: &protocol::ed25519::Pair,
+        recipient: protocol::ed25519::Public,
+        value: Balance,
+        fee: Balance,
+    ) -> Result<Transaction, error::Error> {
+        // Prepare and submit transfer transaction.
+        let transfer_message = protocol::message::Transfer {
+            recipient,
+            balance: value,
+        };
+        let transfer_tx = self
+            .new_signed_transaction(author, transfer_message, fee)
+            .await?;
+        let applied = self.client.submit_transaction(transfer_tx).await?.await?;
+        applied.result?;
+        let block = self.client.block_header(applied.block).await?;
+
+        Ok(Transaction::confirmed(
+            Hash(applied.tx_hash),
+            block.number,
+            Message::Transfer {
+                recipient,
+                balance: value,
+            },
+            fee,
+        ))
+    }
+
+    async fn transfer_from_org(
+        &self,
+        author: &protocol::ed25519::Pair,
+        org_id: Id,
+        recipient: protocol::ed25519::Public,
+        value: Balance,
+        fee: Balance,
+    ) -> Result<Transaction, error::Error> {
+        // Prepare and submit transfer transaction.
+        let transfer_message = protocol::message::TransferFromOrg {
```

Comment:
It seems like there's duplication between these two functions where the only difference is only the messages. Could we pass those in as parameters and have a generic `transfer` function?

Could probably have a `From<TransferFromOrg> for Message` and `From<Transfer> for Message` impls to do the conversion in the call to `confirmed`.

Reply:
To give an update we tried having:
```rust
pub enum Transfer {
  User(protocol::message::Transfer),
  Org(protocol::message::TransferFromOrg),
}
```

The issue was that we couldn't implement `protocol::Message` for this, nor could we return a `Box<dyn protocol::Message>` because one of the methods didn't take `self` -- `result_from_events`. We did our best :(